modules/cache/src/main/java/org/eclipse/xpanse/modules/cache/CaffeineCacheConfig.java

```diff
@@ -56,4 +69,32 @@ private Cache<Object, Object> getServiceFlavorPriceCache() {
                 : DEFAULT_CACHE_EXPIRE_TIME_IN_MINUTES;
         return Caffeine.newBuilder().expireAfterWrite(duration, TimeUnit.MINUTES).build();
     }
+
+
+    private Cache<Object, Object> getCredentialsCache() {
+        Cache cache = Caffeine.newBuilder()
+                .expireAfter(new CredentialCaffeineCacheExpiry())
+                .removalListener((CredentialCacheKey key,
+                                  AbstractCredentialInfo credential,
+                                  RemovalCause cause) -> {
+                    if (Objects.nonNull(key)) {
+                        log.info("Credential cache removed, key: {}, cause: {}", key, cause);
+                    }
+                })
+                .build();
+        return cache;
+    }
+
+    private Cache<Object, Object> getMonitorMetricsCache() {
+        Cache cache = Caffeine.newBuilder()
```

Comment:
we can directly return the cache object here.

Reply:
Updated.